DQM/SiPixelPhase1Common/src/SiPixelCoordinates.cc

```diff
@@ -0,0 +1,916 @@
+// -*- C++ -*-
```

Comment:
hi @schneiml - dqm looks like the wrong spot for such code. Did you consider other users of this?

Second, did you do some study to justify the caching layer? Given the simplicity of the calculations I would think that searching a big map is both slower and more memory intensive than just redoing the calculation.